src/rover.py

```diff
@@ -49,111 +66,68 @@ def joystickToSteeringAngle(self, driving_command, steering_command):
-                steering_angles[self.RR] = 0
-            elif(-80 < deg < -100):
-                # Drive straight backwards
-                steering_angles[self.FL] = 0
-                steering_angles[self.FR] = 0
-                steering_angles[self.CR] = 0
-                steering_angles[self.CL] = 0
-                steering_angles[self.RL] = 0
-                steering_angles[self.RR] = 0
-            elif(100 < deg <= 180):
-                # Drive right forwards
-                steering_angles[self.FL] = 45
-                steering_angles[self.FR] = 45
-                steering_angles[self.CR] = 0
-                steering_angles[self.CL] = 0
-                steering_angles[self.RL] = -45
-                steering_angles[self.RR] = -45
-            elif(-100 > deg >= -180):
-                # Drive right backwards
-                steering_angles[self.FL] = 45
-                steering_angles[self.FR] = 45
-                steering_angles[self.CR] = 0
-                steering_angles[self.CL] = 0
-                steering_angles[self.RL] = -45
-                steering_angles[self.RR] = -45
-            elif(80 > deg >= 0):
-                # Drive left forwards
-                steering_angles[self.FL] = -45
-                steering_angles[self.FR] = -45
-                steering_angles[self.CR] = 0
-                steering_angles[self.CL] = 0
-                steering_angles[self.RL] = 45
-                steering_angles[self.RR] = 45
-            elif(0 > deg > -80):
-                # Drive left backwards
-                steering_angles[self.FL] = -45
-                steering_angles[self.FR] = -45
-                steering_angles[self.CR] = 0
-                steering_angles[self.CL] = 0
-                steering_angles[self.RL] = 45
-                steering_angles[self.RR] = 45
-
-            return steering_angles
+        
         if(self.locomotion_mode == LocomotionMode.ACKERMANN.value):
 
             # No steering if robot is not driving
             if(driving_command is 0):
                 return steering_angles
 
-            # Scale between min and max Ackermann radius
+            # Rear: Scale between min and max Ackermann radius
             if math.cos(math.radians(steering_command)) == 0:
-                r = self.ackermann_r_max
+                rr = self.ackermann_r_max
             else:
-                r = self.ackermann_r_max - \
+                rr = self.ackermann_r_max - \
                     abs(math.cos(math.radians(steering_command))) * \
                     ((self.ackermann_r_max-self.ackermann_r_min))
 
+            # Front: Scale between min and max Ackermann radius
+            if math.cos(math.radians(steering_command)) == 0:
+                fr = self.ackermann_r_max
+            else:
+                fr = self.ackermann_r_max - \
+                    abs(math.cos(math.radians(steering_command))) * \
+                    ((self.ackermann_r_max-self.ackermann_r_min))
```

Comment:
Since we have selected a single maximum radius and a single minimum radius, this scaling does not need to be done to both rear and front wheels. A single scaling between max and min radius simplifies the code and removes one of the scaling variables (like it is done in the `joystickToVelocity`).